### ISSUE TITLE
TextField/TextArea: uses color alone to render focus state in with forced-colors: active

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#5931]
### Provide a general summary of the issue here

With `forced-colors: active`, the TextField and TextArea components fail to render the `box-shadow` style, which adds an additional 1px to the border, around the input. This result in a failure of WCAG success criterion [1.4.1 Use of Color](https://www.w3.org/TR/WCAG22/#use-of-color), because color alone is used to show that the input has focus. 

### 🤔 Expected Behavior?

With `forced-colors: active`, the TextField and TextArea components should to render the `box-shadow` style, which adds an additional 1px to the border, around the input. 

This is caused by the focus-ring css mixin failing to add `forced-colors-adjust: none` to the `:after` element with `@media (forced-colors: active)`:

In https://github.com/adobe/react-spectrum/blob/main/packages/%40adobe/spectrum-css-temp/components/commons/focus-ring.css, we need to add the following: 

```css
@media (forced-colors: active) {
  .spectrum-FocusRing,
  .spectrum-FocusRing-ring,
  .spectrum-FocusRing--quiet {
      --spectrum-high-contrast-focus-ring-color: xvar(--spectrum-high-contrast-focus-ring-color, Highlight);
    &:after {
      forced-color-adjust: none;
    }
  }
}
```

### 😯 Current Behavior

The focus-ring css mixin fails to add `forced-colors-adjust: none` to the `:after` element with `@media (forced-colors: active)` in https://github.com/adobe/react-spectrum/blob/main/packages/%40adobe/spectrum-css-temp/components/commons/focus-ring.css.

### 💁 Possible Solution

In https://github.com/adobe/react-spectrum/blob/main/packages/%40adobe/spectrum-css-temp/components/commons/focus-ring.css, we need to add the following: 

```css
@media (forced-colors: active) {
  .spectrum-FocusRing,
  .spectrum-FocusRing-ring,
  .spectrum-FocusRing--quiet {
    --spectrum-high-contrast-focus-ring-color: xvar(--spectrum-high-contrast-focus-ring-color, Highlight);
    &:after {
      forced-color-adjust: none;
    }
  }
}
```

### 🔦 Context

Accessibility review.

### 🖥️ Steps to Reproduce

1. Open https://react-spectrum.adobe.com/react-spectrum/TextField.html in Google Chrome or Microsoft Edge
2. Using DevTools, set Rendering > Emulate CSS media feature forced-colors to forced-colors: active. It may also be helpful to set Rendering > Emulate CSS media feature prefers-color-scheme to prefers-color-scheme: dark, which will make the Highlight color a more pronounced cyan, instead than the purple that renders with forced-colors: active and prefers-color-scheme: light.
3. Set keyboard focus to the first [TextField Example](https://react-spectrum.adobe.com/react-spectrum/TextField.html#example).
4. Notice that, while the input border-color changes to use the Highlight color, the box-shadow that is usually added the make the border thicker is not rendered in with `forced-colors: active`, and the border remains 1px wide.

### Version

3.0.0

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### What operating system are you using?

macOS Sonoma 14.4 Beta (23E5196e)

### 🧢 Your Company/Team

Adobe/Accessibility
